cmd/preflight/cmd/check_container.go

```diff
@@ -103,85 +107,124 @@ func checkContainerRunE(cmd *cobra.Command, args []string, runpreflight runPrefl
 		return fmt.Errorf("invalid configuration: %w", err)
 	}
 
-	artifactsWriter, err := artifacts.NewFilesystemWriter(artifacts.WithDirectory(cfg.Artifacts))
+	manifestBytes, err := crane.Manifest(containerImage)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not retrieve manifest: %w", err)
+	}
+	manifest, err := cranev1.ParseManifest(bytes.NewBuffer(manifestBytes))
+	if err != nil {
+		return fmt.Errorf("could not parse manifest: %w", err)
 	}
 
-	// Add the artifact writer to the context for use by checks.
-	ctx = artifacts.ContextWithWriter(ctx, artifactsWriter)
+	platformChanged := cmd.Flags().Lookup("platform").Changed
 
-	formatter, err := formatters.NewByName(formatters.DefaultFormat)
-	if err != nil {
-		return err
+	containerImagePlatforms := []string{cfg.Platform}
+
+	// If platform param is not changed, it means that a platform was not specified on the
+	// command line. Therefore, we should process all platforms in the manifest list.
+	// As long as what is poinged to is a manifest list. Otherwise, it will just be the
+	// currnt runtime platform.
+	if isImageList(manifest.MediaType) && !platformChanged {
+		logger.V(log.DBG).Info("manifest list detected, checking all platforms in manifest")
+		manifest, err := cranev1.ParseIndexManifest(bytes.NewBuffer(manifestBytes))
+		if err != nil {
+			return fmt.Errorf("could not parse index manifest: %w", err)
+		}
+		// Preflight was given a manifest list. --platform was not specified.
+		// Therefore, all platforms in the manifest list should be processed.
+		// Create a new slice since the original was for a single platform.
+		containerImagePlatforms = make([]string, 0, len(manifest.Manifests))
+		for _, img := range manifest.Manifests {
+			containerImagePlatforms = append(containerImagePlatforms, img.Platform.Architecture)
+		}
 	}
 
-	opts := generateContainerCheckOptions(cfg)
+	for _, platform := range containerImagePlatforms {
+		logger.Info(fmt.Sprintf("running checks for %s for platform %s", cfg.Image, platform))
```

Comment:
cfg.Image in not getting printed here time="2023-07-19T17:17:14+05:30" level=info msg="running checks for  for platform amd64"